### PR TITLE
Unity docs enhancement

### LINKED ISSUE #208: Unity Documentation
- [ ] Improve Unity Documentation
- [ ] Link to Unity examples
- [ ] E2E Tutorial of working with Unity

### PR DESCRIPTION
**Pull request by cristianFleita**
- [x] Pre-requisites
- [x] Concepts definitions
- [x] Integrating dojo.unity sdk in unity
- [x] Example using dojo starter 
- [x] Common problems section

Resolves #208 
Resolves DOJ-193

### CODE REVIEW

File: docs/pages/client/sdk/unity/important-concepts.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Dojo Unity Concepts
+
+To dive into the exciting world of on-chain games and worlds with Unity, let's explore some essential concepts:
```
Comment (ponderingdemocritus) [inline docs/pages/client/sdk/unity/important-concepts.md:3]: use `onchain` everything instead of on-chain

**Comment by ponderingdemocritus** [comment]
@Larkooo could you review this

File: docs/pages/client/sdk/unity/important-concepts.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+* Control synchronization: Set the maximum number of entities to synchronize.
+* Event-driven communication:
+  * OnSynchronized: Notifies you when entities were successfully synchronized from Dojo world to Unity.
+  * OnEntitySpawned: Triggered whenever a new entity is spawned in the Unity environment.
+
+* Dynamic entity management:
+  * SynchronizeEntities: Asynchronously retrieves and spawns entities from the Dojo world in the Unity environment.
+  * HandleEntityUpdate: Dynamically updates existing entities or spawns new ones based on changes received from the Dojo world, ensuring seamless synchronization.
+
+## Models
```
Comment (Larkooo) [inline docs/pages/client/sdk/unity/important-concepts.md:37]: users dont need to define models themselves so maybe highlight that models can be generated using sozo build --unity

**Comment by cristianFleita** [comment]
@Larkooo @ponderingdemocritus This PR is ready for review!

**Comment by Larkooo** [comment]
Looks good to me for now! I think it's a pretty comprehensive guide for the current state of the dojo.unity SDK. We might wanna review some of the stuff like when we mentions versions and errors further down the road when dojo.unity syncs with dojo versions / updates to the new major dojo version

**Review by Larkooo — APPROVED** [review]

**Pull request merged**